hw/dv/dpi/jtagdpi/jtagdpi.c

```diff
@@ -314,25 +137,21 @@ void *jtagdpi_create(const char *display_name, int listen_port) {
       "  interface remote_bitbang\n"
       "  remote_bitbang_host localhost\n"
       "  remote_bitbang_port %d\n",
-      ctx->display_name, ctx->listen_port, ctx->listen_port);
+      ctx->sock.display_name, ctx->sock.listen_port, ctx->sock.listen_port);
 
   return (void *)ctx;
-
-err_cleanup_return:
-  free(ctx->display_name);
-  free(ctx);
-  return NULL;
 }
 
 void jtagdpi_close(void *ctx_void) {
   struct jtagdpi_ctx *ctx = (struct jtagdpi_ctx *)ctx_void;
   if (!ctx) {
     return;
   }
-  tcp_server_client_close(ctx);
-  tcp_server_stop(ctx);
+  // Kill the socket thread
```

Comment:
Kill -> Shut down (there are ways to "cancel" or "kill" a thread, but that's not what we're doing here)